Jakarta EE, Quarkus 3 and Hibernate ORM upgrade documentation

mabartos opened the pull request:
Closes #16251

![image](https://github.com/keycloak/keycloak/assets/38039883/1c56183e-b5c3-46e5-907a-1dd52b2c9892)

![image](https://github.com/keycloak/keycloak/assets/38039883/19891236-3b4e-45c5-bc07-7ff51adbed7e)
![image](https://github.com/keycloak/keycloak/assets/38039883/0f832e5b-9896-4027-98f1-a5d2a720d64f)

@vmuzikar @andymunro @ahus1 @keycloak/store  Could you please check it? Thanks

mabartos committed:
Jakarta EE and Quarkus 3 upgrade documentation

Closes #16251

Co-authored-by: Václav Muzikář <vaclav@muzikari.cz>
Co-authored-by: Alexander Schwartz <aschwart@redhat.com>

mabartos commented:
Documentation job failures are not related to this PR. They were resolved by: https://github.com/keycloak/keycloak/issues/20259

vmuzikar commented on docs/documentation/release_notes/topics/22_0_0.adoc:
```diff
@@ -1,3 +1,27 @@
+Quarkus 3 continues the tradition of propelling Java development by moving fast and providing a cutting-edge user experience with the latest technologies.
+It continues to improve overall performance and efficiency.
+
+It provides specific breaking changes, so check the migration guide for details.
```
I'd maybe slightly rephrase.
```suggestion
Since Quarkus 3 is a major upgrade, some breaking changes were inevitable. See the migration guide for details.
```

vmuzikar commented on docs/documentation/release_notes/topics/22_0_0.adoc:
```diff
@@ -1,3 +1,27 @@
+
+It provides specific breaking changes, so check the migration guide for details.
+
+= We use Hibernate ORM 6 now
+
+Keycloak now benefits from the upgrade to Hibernate ORM 6.2, which includes improved performance, better SQL, modern JDK support, and support for modern RDBMS features.
+
+If you have custom providers or JPA entities, these changes may affect you.
+
+See the migration guide for more details.
```
I think we should ask @keycloak/store team for a review of this.

vmuzikar commented on docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc:
```diff
@@ -1,3 +1,52 @@
+
+=== Stop using `quarkus.hibernate-orm.*` properties
+For Quarkus 3, it is impossible to simultaneously specify Hibernate ORM properties for datasources via the `persistence.xml` file and Quarkus properties.
+As for Keycloak, these properties are specified via the `persistence.xml` file, and it is required to specify the additional properties there.
```
I believe, we still plan to address this for 22: https://github.com/keycloak/keycloak/issues/19427
@ahus1 Can you please confirm?

mabartos replied:
I'd keep the information here as it reflects the current state. If the issue is solved for KC 22, we can remove it from the migration guide. @vmuzikar @ahus1 WDYT?

ahus1 replied:
@mabartos +1 - if we choose to update it and find the time to address it, the docs will change. For now, I'd rather have the docs in main match the state we have in main - and this is what @mabartos prepared here.

ahus1 commented:
Will have a look later today.

ahus1 reviewed (changes requested):
Thank you for adding these docs! I created a PR with two commits, see https://github.com/mabartos/keycloak/pull/168 for details. Please ping me again when I should re-review it.

mabartos commented:
@vmuzikar @ahus1 I've added your suggestions. Thanks! The formatting preview in the description of this PR is also updated, so you chan check the provided changes. 

Ready for another round of reviews

ahus1 reviewed (approved):
LGTM!
